lua/entities/kleiner/sv_spawning.lua

```diff
@@ -0,0 +1,93 @@
+﻿-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+-- INSTALL: CINEMA
+CreateConVar("kleiner_spawncount", "18", FCVAR_ARCHIVE + FCVAR_PROTECTED, "Number of Kleiner NPCs to spawn automatically.", 0, 100)
+KLEINER_NPCS = KLEINER_NPCS or {}
+KLEINER_NPCS_FILTER = KLEINER_NPCS_FILTER or {}
+KLEINER_NPC_TARGETS = KLEINER_NPC_TARGETS or {}
+KLEINER_NPCS_CURRENT_NUMBER = KLEINER_NPCS_CURRENT_NUMBER or 0
+KLEINER_DESIRED_NUMBER = KLEINER_DESIRED_NUMBER or GetConVar("kleiner_spawncount"):GetInt() or 0
+
+function KLEINER_NPC_SETUPTIMER(value)
+    KLEINER_DESIRED_NUMBER = value
+    timer.Destroy("kleiner_spawner")
+
+    if (KLEINER_DESIRED_NUMBER > 0) then
+        timer.Create("kleiner_spawner", 1, 0, function()
+            if (table.Count(KLEINER_NPCS) < KLEINER_DESIRED_NUMBER) then
+                local newkleiner = ents.Create("kleiner")
+
+                if (IsValid(newkleiner)) then
+                    local spawnpoint = gmod.GetGamemode().SpawnPoints and table.Random(gmod.GetGamemode().SpawnPoints):GetPos() or Vector(0, 0, 16)
```

Comment:
This line threw a NULL entity error

Reply:
probably just ents.FindByClass("info_player_start") then cache it, and if you find a NULL entity, update the cache